jakarta-eclipse/src/org/jakarta/jdt/BeanValidationDiagnosticsCollector.java

```diff
@@ -0,0 +1,326 @@
+					!fieldType.equals(getSignatureFormatOfType(CHAR_SEQUENCE)) &&
+					!fieldType.equals(getSignatureFormatOfType(BYTE)) &&
+					!fieldType.equals(getSignatureFormatOfType(SHORT)) &&
+					!fieldType.equals(getSignatureFormatOfType(INTEGER)) &&
+					!fieldType.equals(getSignatureFormatOfType(LONG)) &&
+					!fieldType.equals(Signature.SIG_BYTE) &&
+					!fieldType.equals(Signature.SIG_SHORT) &&
+					!fieldType.equals(Signature.SIG_INT) &&
+					!fieldType.equals(Signature.SIG_LONG)) {
+				
+				diagnostics.add(completeDiagnostic(new Diagnostic(fieldAnnotationrange,
+						"The @" + annotation.getElementName() + " annotation can only be used on: \n- BigDecimal \n- BigInteger \n- CharSequence"
+						+ "\n- byte, short, int, long (and their respective wrappers) \n type fields.")));	
+			}
+		} else if (annotation.getElementName().equals(EMAIL)) {
+
+			if (	!fieldType.equals(getSignatureFormatOfType(STRING)) &&
+					!fieldType.equals(getSignatureFormatOfType(CHAR_SEQUENCE))) {
+				
+				diagnostics.add(completeDiagnostic(new Diagnostic(fieldAnnotationrange,
+						"The @" + annotation.getElementName() + " annotation can only be used on String and CharSequence type fields.")));	
+			}
+		} else if (annotation.getElementName().equals(FUTURE)|| annotation.getElementName().equals(FUTURE_OR_PRESENT) 
+				|| annotation.getElementName().equals(PAST) || annotation.getElementName().equals(PAST_OR_PRESENT)) {
+
+			if (	!fieldType.equals(getSignatureFormatOfType(DATE)) &&
+					!fieldType.equals(getSignatureFormatOfType(CALENDAR)) &&
+					!fieldType.equals(getSignatureFormatOfType(INSTANT)) &&
+					!fieldType.equals(getSignatureFormatOfType(LOCAL_DATE)) &&
+					!fieldType.equals(getSignatureFormatOfType(LOCAL_DATE_TIME)) &&
+					!fieldType.equals(getSignatureFormatOfType(LOCAL_TIME)) &&
+					!fieldType.equals(getSignatureFormatOfType(MONTH_DAY)) &&
+					!fieldType.equals(getSignatureFormatOfType(OFFSET_DATE_TIME)) &&
+					!fieldType.equals(getSignatureFormatOfType(OFFSET_TIME)) &&
+					!fieldType.equals(getSignatureFormatOfType(YEAR)) &&
+					!fieldType.equals(getSignatureFormatOfType(YEAR_MONTH)) &&
+					!fieldType.equals(getSignatureFormatOfType(ZONED_DATE_TIME)) &&
+					!fieldType.equals(getSignatureFormatOfType(HIJRAH_DATE)) &&
+					!fieldType.equals(getSignatureFormatOfType(JAPANESE_DATE)) &&
+					!fieldType.equals(getSignatureFormatOfType(MINGUO_DATE)) &&
+					!fieldType.equals(getSignatureFormatOfType(THAI_BUDDHIST_DATE))) {
+				
+				diagnostics.add(completeDiagnostic(new Diagnostic(fieldAnnotationrange,
+						"The @" + annotation.getElementName() + " annotation can only be used on: - Date - Calendar - Instant"
+						+ "- LocalDate - LocalDateTime - LocalTime - MonthDay - OffsetDateTime "
+						+ "- OffsetTime - Year - YearMonth - ZonedDateTime - "
+						+ "HijrahDate - JapaneseDate - JapaneseDate - MinguoDate - "
+						+ "ThaiBuddhistDate type fields.")));	
```

Comment:
If this diagnostic message is staying as one line, we can remove the dashes and just use commas.